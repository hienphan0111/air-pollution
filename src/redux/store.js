import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries';
import citiesReducer from './cities';
// import cityReducer from '@/routes/city-detail';
import cityReducer from './city';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    cities: citiesReducer,
    cityInfo: cityReducer,
  },
});

export default store;
