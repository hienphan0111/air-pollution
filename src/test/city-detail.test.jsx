/**
 * @jest-environment jsdom
 */

import React from 'react';
import 'mock-react-redux';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect.js';
import { Provider, useSelector, useDispatch } from 'react-redux';
import store from '../redux/store';
import CityDetail from '../routes/city-detail';
import axios from 'axios';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import mockReactRedux from 'mock-react-redux';

jest.mock('axios', () => ({
  post: jest.fn(),
  get: jest.fn(),
}));

const reactRedux = { useDispatch, useSelector };

describe('Render the detail info of city', () => {
  
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector')
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch')

  const cityDetail = {
    id: 5,
    name: 'one',
    aqi: 10,
    time: {
      s: '2022-21-12 17:00'
    },
    forecast: [{
      day: '2023-03-14',
      avg: 133,
      min: 90,
      max: 160,
    }]
  }
  
  it('Render detail info of city', () => {

    mockReactRedux().state({
      cityInfo: {
        cityDetail,
      }
    });

    const detail = TestRenderer.create(
      <MemoryRouter>
        <CityDetail />
      </MemoryRouter>).toJSON();

    expect(detail).toMatchSnapshot();
  })

});