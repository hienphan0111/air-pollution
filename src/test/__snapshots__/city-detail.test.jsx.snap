// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the detail info of city Render detail info of city 1`] = `
<div
  className="city-container"
>
  <div
    className="bar-head"
  >
    <div
      className="nav"
    >
      <a
        href="/cities"
        onClick={[Function]}
      >
        <svg
          className="w-6 h-6 back-icon"
          fill="none"
          stroke="currentColor"
          strokeWidth={1.5}
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 19.5L8.25 12l7.5-7.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </a>
      <span>
        2023
      </span>
    </div>
    <span>
      AIR QUALITY DETAIL
    </span>
    <div
      className="setting-icon"
    >
      <svg
        className="w-6 h-6"
        fill="none"
        stroke="currentColor"
        strokeWidth={1.5}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 18.75a6 6 0 006-6v-1.5m-6 7.5a6 6 0 01-6-6v-1.5m6 7.5v3.75m-3.75 0h7.5M12 15.75a3 3 0 01-3-3V4.5a3 3 0 116 0v8.25a3 3 0 01-3 3z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
      <svg
        className="w-6 h-6"
        fill="none"
        stroke="currentColor"
        strokeWidth={1.5}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.343 3.94c.09-.542.56-.94 1.11-.94h1.093c.55 0 1.02.398 1.11.94l.149.894c.07.424.384.764.78.93.398.164.855.142 1.205-.108l.737-.527a1.125 1.125 0 011.45.12l.773.774c.39.389.44 1.002.12 1.45l-.527.737c-.25.35-.272.806-.107 1.204.165.397.505.71.93.78l.893.15c.543.09.94.56.94 1.109v1.094c0 .55-.397 1.02-.94 1.11l-.893.149c-.425.07-.765.383-.93.78-.165.398-.143.854.107 1.204l.527.738c.32.447.269 1.06-.12 1.45l-.774.773a1.125 1.125 0 01-1.449.12l-.738-.527c-.35-.25-.806-.272-1.203-.107-.397.165-.71.505-.781.929l-.149.894c-.09.542-.56.94-1.11.94h-1.094c-.55 0-1.019-.398-1.11-.94l-.148-.894c-.071-.424-.384-.764-.781-.93-.398-.164-.854-.142-1.204.108l-.738.527c-.447.32-1.06.269-1.45-.12l-.773-.774a1.125 1.125 0 01-.12-1.45l.527-.737c.25-.35.273-.806.108-1.204-.165-.397-.505-.71-.93-.78l-.894-.15c-.542-.09-.94-.56-.94-1.109v-1.094c0-.55.398-1.02.94-1.11l.894-.149c.424-.07.765-.383.93-.78.165-.398.143-.854-.107-1.204l-.527-.738a1.125 1.125 0 01.12-1.45l.773-.773a1.125 1.125 0 011.45-.12l.737.527c.35.25.807.272 1.204.107.397-.165.71-.505.78-.929l.15-.894z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <path
          d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </div>
  </div>
  <div
    className="detail-body"
  >
    <div
      className="ct-header"
    >
      <h3>
        one
      </h3>
    </div>
    <div
      className="ct-info"
    >
      <div
        className="aqi"
      >
        <span>
          2022-21-12 17:00
        </span>
        <span
          className="aqi-n"
          style={
            {
              "backgroundColor": "#0c563b",
            }
          }
        >
          10
        </span>
        <span
          className="aq-title"
        >
          Good
        </span>
      </div>
      <div
        className="forecast"
      >
        <h4>
          PM25 Forecast
        </h4>
        <div
          className="fc-table"
        >
          <div
            className="fc-head"
          >
            <span>
              Date
            </span>
            <span>
              avg
            </span>
            <span>
              min
            </span>
            <span>
              max
            </span>
          </div>
          <div
            className="pm25"
          >
            <span>
              2023-03-14
            </span>
            <span>
              133
            </span>
            <span>
              90
            </span>
            <span>
              160
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
